File: leetcode/medium/128_longest_consecutive_sequence/longest_consecutive_sequence.py
from typing import List


class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        values = set(nums)
        longest = 0
        current = 0
        for key in values:
            if key - 1 in values:
                continue
            current = 0
            while key in values:
                key += 1
                current += 1
            longest = max(longest, current)
        return longest

File: leetcode/medium/128_longest_consecutive_sequence/longest_consecutive_sequence.cpp
#include <unordered_set>
#include <vector>

class Solution {
public:
  int longestConsecutive(std::vector<int> &nums) {
    std::unordered_set<int> nums_set(nums.begin(), nums.end());
    int max = 0;
    for (auto num : nums) {
      if (nums_set.find(num - 1) != nums_set.end()) {
        continue;
      }
      int curr = num;
      while (nums_set.find(curr) != nums_set.end()) {
        curr++;
      }
      max = std::max(max, curr - num);
    }
    return max;
  }
};
